pkg/generators/rules/omitempty_match_case.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rules
+
+import (
+	"reflect"
+	"strings"
+
+	"k8s.io/gengo/types"
+)
+
+/*
+OmitEmptyMatchCase implements APIRule interface.
+"omitempty" must appear verbatim (no case variants).
+*/
+type OmitEmptyMatchCase struct{}
+
+// Name returns the name of APIRule
+func (n *OmitEmptyMatchCase) Name() string {
+	return "omitempty_match_case"
+}
+
+// Validate evaluates API rule on type t and returns a list of field names in
+// the type that violate the rule. Empty field name [""] implies the entire
+// type violates the rule.
+func (n *OmitEmptyMatchCase) Validate(t *types.Type) ([]string, error) {
```

Comment:
I'm assuming we don't have to nil-check?

Reply:
yeah, type is always non-nil